src/sage/schemes/elliptic_curves/hom_composite.py

```diff
@@ -893,3 +896,291 @@ def inseparable_degree(self):
             1331
         """
         return prod(phi.inseparable_degree() for phi in self._phis)
+
+    def is_cyclic(self, _check_j=True, _reduce_kernel=True):
+        r"""
+        Determine whether the isogeny is cyclic (separable with cyclic kernel).
+
+        ALGORITHM:
+
+        For each prime $\ell$ with $\ell^2$ dividing the isogeny degree,
+        test that $\ell$-torsion is not fully nullified. This is done by
+        evaluating the isogeny modulo the division polynomial of the domain curve,
+        reducing the modulus on the way when parts of the torsion get nullified.
+
+        For each $\ell$, only the smallest segment of isogenies is considered,
+        such that it includes all isogenies whose degrees are divisible by $\ell$.
+
+        INPUT: The following optimization flags are intended only for debugging
+        and testing.
+
+        If `_check_j` is set, then the segment is further shrinked by eliminating
+        prefix/suffix chains of $\ell$-isogenies via j-invariant checks. In many
+        applications, the isogeny degrees would be ordered, so that only
+        j-invariant checks would be performed (unless there is a "backtracking"
+        j-invariant step, which is not necessarily the dual).
+
+        If `_reduce_kernel` is set, then, after the first step in the segment,
+        the modulus is replaced by the kernel polynomial of the dual isogeny,
+        which has degree (l-1)/2 instead of (l^2-1)/2.
+
+        EXAMPLES::
+
+            sage: E0 = EllipticCurve(GF((2**64+13)**2), [4, 0])
+
+            sage: phi1 = E0.isogenies_prime_degree(2)[0]
+            sage: (phi1.dual() * phi1).is_cyclic()
+            False
+
+            sage: phi2_list = phi1.codomain().isogenies_prime_degree(2)
+            sage: sorted((phi2 * phi1).is_cyclic() for phi2 in phi2_list)
+            [False, True, True]
+
+        Works even if the cyclicity is broken across more than 2 isogenies:
+        (e.g., when commuting isogenies are swapped: degrees 2-2-5 <-> 2-5-2)
+
+            sage: phi2 = phi1.codomain().isogenies_prime_degree(5)[0]
+            sage: phi3_list = phi2.codomain().isogenies_prime_degree(2)
+            sage: sorted((phi3 * phi2 * phi1).is_cyclic() for phi3 in phi3_list)
+            [False, True, True]
+
+        "Recursive" composite isogenies are not flattened, but should still
+        return correct results:
```

Comment:
```suggestion
        Endomorphisms as composite isogenies are not flattened but still returns correct results::
```
I think this is what you mean.

Reply:
Hmm, no, I meant EllipticCurveHom_composite where one or more internal isogenies are also a EllipticCurveHom_composite instances.

Reply:
Ahh okay. I think “recursive” is still not the correct word, or it should at least be defined somewhere somewhat formally. Also the double colon at the end :D

Reply:
How about "nested"?